cpp/scripts/run-clang-format.py

```diff
@@ -19,41 +19,15 @@
 import os
 import subprocess
 import argparse
-import gitutils
 
 
-def listAllSourceFiles(fileRegex, srcdirs, dstdir, inplace):
-    allFiles = []
-    for srcdir in srcdirs:
-        for root, dirs, files in os.walk(srcdir):
-            for f in files:
-                if re.search(fileRegex, f):
-                    src = os.path.join(root, f)
-                    if inplace:
-                        _dir = root
-                    else:
-                        _dir = os.path.join(dstdir, os.path.basename(root))
-                    dst = os.path.join(_dir, f)
-                    allFiles.append((src, dst))
-    return allFiles
-
-
-def listAllChangedFiles(fileRegex, dstdir, inplace):
-    allFiles = []
-
-    def checkThisFile(f):
-        return True if fileRegex.search(f) else False
-
-    files = gitutils.modifiedFiles(filter=checkThisFile)
-    for f in files:
-        dst = f if inplace else os.path.join(dstdir, f)
-        allFiles.append((f, dst))
-    return allFiles
+VersionRegex = re.compile(r"clang-format version ([0-9.]+)")
 
 
-def parseArgs():
+def parse_args():
     argparser = argparse.ArgumentParser("Runs clang-format on a project")
-    argparser.add_argument("-dstdir", type=str, default=".",
+    argparser.add_argument("-dstdir", type=str,
+                           default="/tmp/cuml-clang-format",
                            help="Path to the current build directory.")
```

Comment:
The help text seems inconsistent with the default? Maybe it should be "Output path for formatted results"

Reply:
Should default to None and create a new temporary dir with `mkdtemp`. If multiple users are running on a shared lab machine, having a global shared build path could cause trouble

Reply:
Good point. I have updated the code to use `mkdtemp`.